src/PIL/DdsImagePlugin.py

```diff
@@ -124,114 +335,144 @@ def _open(self):
+                raise OSError(msg)
+            rawmode = rawmode[::-1]
+        elif pfflags & DDPF.ALPHA:
+            if bitcount == 8:
+                self._mode = "L"
+            else:
+                msg = f"Unsupported bitcount {bitcount} for {pfflags}"
+                raise OSError(msg)
+        elif pfflags & DDPF.LUMINANCE:
+            if bitcount == 8:
+                self._mode = "L"
+            elif bitcount == 16 and pfflags & DDPF.ALPHAPIXELS:
+                self._mode = "LA"
+            else:
+                msg = f"Unsupported bitcount {bitcount} for {pfflags}"
+                raise OSError(msg)
+        elif pfflags & DDPF.PALETTEINDEXED8:
             self._mode = "P"
             self.palette = ImagePalette.raw("RGBA", self.fp.read(1024))
-            self.tile = [("raw", (0, 0) + self.size, 0, "L")]
-        else:
-            data_start = header_size + 4
-            n = 0
-            if fourcc == b"DXT1":
+        elif pfflags & DDPF.FOURCC:
+            data_offs = header_size + 4
+            if fourcc == D3DFMT.DXT1:
+                self._mode = "RGBA"
                 self.pixel_format = "DXT1"
                 n = 1
-            elif fourcc == b"DXT3":
+            elif fourcc == D3DFMT.DXT3:
+                self._mode = "RGBA"
                 self.pixel_format = "DXT3"
                 n = 2
-            elif fourcc == b"DXT5":
+            elif fourcc == D3DFMT.DXT5:
+                self._mode = "RGBA"
                 self.pixel_format = "DXT5"
                 n = 3
-            elif fourcc == b"ATI1":
+            elif fourcc in (D3DFMT.BC4U, D3DFMT.ATI1):
+                self._mode = "L"
                 self.pixel_format = "BC4"
                 n = 4
-                self._mode = "L"
-            elif fourcc in (b"ATI2", b"BC5U"):
-                self.pixel_format = "BC5"
-                n = 5
+            elif fourcc == D3DFMT.BC5S:
                 self._mode = "RGB"
-            elif fourcc == b"BC5S":
                 self.pixel_format = "BC5S"
                 n = 5
+            elif fourcc in (D3DFMT.BC5U, D3DFMT.ATI2):
                 self._mode = "RGB"
-            elif fourcc == b"DX10":
-                data_start += 20
+                self.pixel_format = "BC5"
+                n = 5
+            elif fourcc == D3DFMT.DX10:
+                data_offs += 20
                 # ignoring flags which pertain to volume textures and cubemaps
                 (dxgi_format,) = struct.unpack("<I", self.fp.read(4))
                 self.fp.read(16)
-                if dxgi_format in (DXGI_FORMAT_BC5_TYPELESS, DXGI_FORMAT_BC5_UNORM):
+                if dxgi_format in (
+                    DXGI_FORMAT.BC1_UNORM,
+                    DXGI_FORMAT.BC1_UNORM_SRGB,
+                    DXGI_FORMAT.BC1_TYPELESS,
+                ):
+                    self._mode = "RGBA"
+                    self.pixel_format = "BC1"
+                    n = 1
```

Comment:
Our tradition is to add tests for new code, to try and prevent it breaking in the future. Do you have an image that can be used to test this?

Reply:
Ye, but they are from that game and developers do not allow usage of their assets. So I need to craft one

Reply:
Oh, sorry, replied to wrong comment

Reply:
I've added a commit to https://github.com/REDxEYE/Pillow/pull/7. I copied the existing test image dxt1-rgb-4bbp-noalpha_MipMaps-1.dds and modified it to be BC1_UNORM and BC1_TYPELESS. However, I have removed BC1_UNORM_SRGB. See what you think.